glean-core/src/ping/mod.rs

```diff
@@ -67,12 +70,80 @@ impl PingMaker {
         current_seq as usize
     }
 
+    fn get_start_end_times(&self, glean: &Glean, storage_name: &str) -> (Metric, Metric) {
+        let time_unit = TimeUnit::Minute;
+
+        let start_time = DatetimeMetric::new(
+            CommonMetricData {
+                name: format!("{}#start", storage_name),
+                category: "".into(),
+                send_in_pings: vec![INTERNAL_STORAGE.into()],
+                lifetime: Lifetime::User,
+                ..Default::default()
+            },
+            time_unit,
+        );
+
+        let end_time = DatetimeMetric::new(
```

Comment:
Do we need this one? I think we only require the `start_time` one to be stored in memory, `end_time` will always be `Metric::Datetime(fixed_now, time_unit)`